tests/unit/data/data_pipeline/test_round_robin.py

```diff
@@ -76,6 +77,28 @@ def test_op_works_when_pipelines_have_different_lengths(self) -> None:
 
             pipeline.reset()
 
+    @pytest.mark.skipif(
+        python_devel_only(),
+        reason="New fairseq2n API in Python-only installation. Skipping till v0.2.",
```

Comment:
Not for this PR: 

I'm not fond of the behavior of this `python devel only`
First it seems to be about not disturbing Python only developer that didn't build fairseq2n locally. So `fairseq2n_dev_only` would be more clear.
Second I'd prefer if it was annotation `@fairseq2n_dev_only` with  a standard reason.
Third when are we supposed to removed those annotations ? Will we have to do this manually ? 

I don't really see who would want to run our unit tests from github main branch but can't build fairseq2n.

Reply:
> First it seems to be about not disturbing Python only developer that didn't build fairseq2n locally. So fairseq2n_dev_only would be more clear.

I think if you convert it to a standalone annotation, a name like `@fairseq2n_dev_only` makes sense, but as a parameter to `skip_if`, `fairseq2n_dev_only` does not sound right, because it is just the opposite: you want to skip the test *if* this is a python-only installation. I am open to having a standalone annotation though.

> Third when are we supposed to removed those annotations ? Will we have to do this manually ?

There are two ways to resolve this issue. We can either have a version range check, like `@skipif(fairseq2n.__version__< 0.2)` which would not require us to remove the annotation in the future. The disadvantage is, those checks would not be relevant after a stable release and will start to pile up with time for no good reason. The other alternative is what we have right now. Just checking whether the versions do not match `@skipif(fairseq2.__version__ != fairseq2n.__version__)`. The advantage is that this forces you to remove the annotations before a stable release (otherwise the tests won't run), the disadvantage is that it is a manual work that needs to be done as part of the version bump. Having said that I think we won't need this annotation that often in the future once the native library becomes more mature. The manual removal is not ideal, but we might add some extra measures like forcibly fail any test that has this annotation and if the latest version is already past the one in the annotation.

> Can someone explain me the use case ?

Pretty much anyone who wants to contribute to the Python code base of fairseq2, but does not want to deal (or have experience) with C++. The whole reason why we ended up with fairseq2n in last minute was because people had quite a bit of trouble with building the library (e.g. while working on seq-gen and UnitY upstreaming). The original feedback was to completely separate the C++ parts to a separate repo, but we then decided to keep it here and offer pre-built binaries so that people who want to contribute to fairseq2 won't have to build the native parts if they don't want to. By the way, I got this approach pretty much from jax where they similarly separate the library into two parts: jax (python only) and jaxlib (pre-built).

Reply:
Ok, just gave it a try in #83. Let me know what you think.

Reply:
thanks, let's follow on that PR, and merge this one.